llms/mlx_lm/server.py

```diff
@@ -0,0 +1,326 @@
+import argparse
+import json
+import os
+import time
+import uuid
+from collections import namedtuple
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from typing import List, Optional, Tuple
+
+import mlx.core as mx
+import mlx.nn as nn
+import numpy as np
+from transformers import PreTrainedTokenizer
+
+from .utils import load
+
+_model: Optional[nn.Module] = None
+_tokenizer: Optional[PreTrainedTokenizer] = None
+
+
+def load_model(model_path: str, adapter_file: Optional[str] = None):
+    global _model
+    global _tokenizer
+    _model, _tokenizer = load(model_path, adapter_file=adapter_file)
+
+
+StopCondition = namedtuple("StopCondition", ["stop_met", "trim_needed", "trim_length"])
+
+
+def stopping_criteria(
+    tokens: List[int],
+    stop_id_sequences: List[np.ndarray],
+    eos_token_id: int,
+    max_tokens: int,
+) -> StopCondition:
+    """
+    Determines whether a stop condition for text generation has been met.
+
+    It checks if the generated text has reached a specified maximum length,
+    ended with an EOS token, or contains a specified sequence of tokens
+    that should trigger stopping.
+
+    Parameters:
+    - tokens (List[int]): The list of generated token IDs.
+    - stop_id_sequences (List[np.ndarray]): A list of numpy arrays, each representing a sequence
+      of token IDs that, if generated, should trigger the end of text generation.
+    - eos_token_id (int): The token ID representing the end-of-sequence (EOS) marker.
+    - max_tokens (int): The maximum number of tokens to generate. Text generation will stop
+      if this number is reached.
+
+    Returns:
+    - StopCondition: A named tuple with the fields `stop_met` (bool), `trim_needed` (bool),
+      and `trim_length` (int), providing details about the stop condition.
```

Comment:
Nit: Could you change this docstring to use the same format as the others in mlx lm? (`Args:` instead of `Parameters:` and no `-` for each parameter in the list.